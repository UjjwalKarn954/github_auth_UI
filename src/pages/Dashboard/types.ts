export type StateProps = {
    isLoading: boolean;
};

export type ArtifactLogs = {
    name: string;
    output: string;
};

export type Props = StateProps;
