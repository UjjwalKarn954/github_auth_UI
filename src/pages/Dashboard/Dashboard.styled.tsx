import { Card } from '@medly-components/core';
import { defaultTheme } from '@medly-components/theme';
import { styled } from '@styled';
import { Link } from 'react-router-dom';

export const StyledCard = styled(Card)`
    gap: 2rem;
    min-width: 30rem;
`;

export const StyledLink = styled(Link)`
    font-weight: 700;
    color: ${defaultTheme.colors.grey[900]};
`;
