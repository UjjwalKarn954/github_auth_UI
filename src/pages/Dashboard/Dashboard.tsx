import FlexContainer from '@components/FlexContainer';
import { PageContent } from '@components/layout';
import PrimaryButton from '@components/PrimaryButton';
import { Text } from '@medly-components/core';
import { LockIcon, StarRate18pxRoundedIcon } from '@medly-components/icons';
import { defaultTheme } from '@medly-components/theme';
import { fetch } from '@utils';
import React, { useState } from 'react';
import { StyledCard, StyledLink } from './Dashboard.styled';
import { Props } from './types';

export const Dashboard: React.FC<Props> = ({ isLoading }) => {
    const [repositories, setRepositories] = useState([]);

    async function gitLogin() {
        const { response } = await fetch({
            method: 'GET',
            url: 'http://localhost:9000/login/github'
        });
        window.location.assign(response.data);
    }

    async function getRepo() {
        const { response } = await fetch({
            method: 'GET',
            url: 'http://localhost:9000/repo/getRepo',
            params: {
                token: localStorage.getItem('access_token')
            }
        });
        setRepositories(response.data);
    }

    return (
        <PageContent isLoading={isLoading}>
            <FlexContainer flexWrap alignItems="center">
                {repositories.map(repo => (
                    <StyledCard display="flex" key={repo.id}>
                        <Text uppercase textVariant="h4" textWeight="Medium" textColor={defaultTheme.colors.grey[900]}>
                            {repo.name}
                            {repo.private && <LockIcon size="S" iconColor={defaultTheme.colors.blue[700]} />}
                        </Text>
                        <FlexContainer alignItems="center">
                            <FlexContainer alignItems="center">
                                {repo.stargazers_count} <StarRate18pxRoundedIcon size="L" iconColor={defaultTheme.colors.yellow[500]} />
                            </FlexContainer>
                            <StyledLink to={`/${repo.name}/logs`}>Get Logs</StyledLink>
                        </FlexContainer>
                    </StyledCard>
                ))}
            </FlexContainer>
            <FlexContainer minWidth="100%" justifyContent="center">
                <PrimaryButton clickHandler={gitLogin}>Login with Github</PrimaryButton>
                <PrimaryButton clickHandler={getRepo}>Fetch repo</PrimaryButton>
            </FlexContainer>
        </PageContent>
    );
};
