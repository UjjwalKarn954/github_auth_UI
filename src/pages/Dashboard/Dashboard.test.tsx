import { initialState } from '@store';
import { fireEvent, mockAxios, renderWithRouter, screen, waitFor } from '@test-utils';
import React from 'react';
import { Provider } from 'react-redux';
import reduxMockStore from 'redux-mock-store';
import Dashboard from './Dashboard.container';

describe('Dashboard', () => {
    const mockStore = reduxMockStore(),
        renderDashboard = (store: ReturnType<typeof mockStore>) =>
            renderWithRouter(
                <Provider store={store}>
                    <Dashboard />
                </Provider>
            );

    const location = window.location;

    beforeAll(() => {
        delete window.location;
        window.location = { ...location, assign: jest.fn() };
    });

    afterAll(() => {
        window.location = location;
    });

    it('should render properly', () => {
        const store = mockStore({ ...initialState, loading: { dashboard: { isLoading: false } } });
        const { container } = renderDashboard(store);
        expect(container).toMatchSnapshot();
    });

    it('should show loading if isLoading Prop is true', () => {
        const store = mockStore({ ...initialState, loading: { dashboard: { isLoading: true } } });
        const { container } = renderDashboard(store);
        expect(container).toMatchSnapshot();
    });

    it('should get login callback url', async () => {
        mockAxios.onGet('http://localhost:9000/login/github').reply(200, 'https://abc.com');

        const store = mockStore({ ...initialState, loading: { dashboard: { isLoading: true } } });
        renderDashboard(store);

        const loginBtn = screen.getByRole('button', { name: /Login with GitHub/i });
        fireEvent.click(loginBtn);

        waitFor(() => expect(window.location.assign).toHaveBeenCalled(), { timeout: 5000 });
    });

    it('should get repositories', async () => {
        mockAxios.onGet('http://localhost:9000/repo/getRepo').reply(200, [{ id: 1, name: 'repo1', private: true }]);

        mockAxios.onGet('http://localhost:9000/action/runs').reply(200, 'Some random logs');

        const store = mockStore({ ...initialState, loading: { dashboard: { isLoading: true } } });
        const { container } = renderDashboard(store);

        const fetchRepoBtn = screen.getByRole('button', { name: /Fetch repo/i });
        fireEvent.click(fetchRepoBtn);

        const repoData = await screen.findByText('repo1');
        expect(repoData).toBeInTheDocument();

        expect(container.querySelector('svg')).toBeInTheDocument();
    });
});
