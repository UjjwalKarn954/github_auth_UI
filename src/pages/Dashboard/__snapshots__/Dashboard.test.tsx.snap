// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render properly 1`] = `
.c6 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c4 {
  border: none;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  padding: 0.8rem 2.4rem 1rem;
  border-radius: 0.8rem;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4,
.c4 .c5,
.c4 .sc-dlfnbm,
.c4 .sc-dlfnbm * {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c4:hover {
  cursor: pointer;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  cursor: not-allowed;
}

.c4::after {
  content: '';
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: transparent;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.8rem;
  border-style: solid;
}

.c4:disabled {
  color: #dfe4e9;
  background-color: transparent;
}

.c4:disabled::after {
  border-width: 0.1rem;
  border-color: #dfe4e9;
}

.c4:disabled .sc-dlfnbm * {
  fill: #dfe4e9;
}

.c4:not(:disabled):not(:hover) {
  color: #303C48;
  background-color: transparent;
}

.c4:not(:disabled):not(:hover)::after {
  border-width: 0.1rem;
  border-color: #303C48;
}

.c4:not(:disabled):not(:hover) .sc-dlfnbm * {
  fill: #303C48;
}

.c4:not(:disabled):active {
  color: #546A7F;
  background-color: transparent;
}

.c4:not(:disabled):active::after {
  border-width: 0.2rem;
  border-color: #546A7F;
}

.c4:not(:disabled):active .sc-dlfnbm * {
  fill: #546A7F;
}

.c4:not(:disabled):not(:active):hover {
  color: #435465;
  background-color: transparent;
  box-shadow: 0 0.2rem 0.8rem rgba(67,84,101,0.35);
}

.c4:not(:disabled):not(:active):hover::after {
  border-width: 0.2rem;
  border-color: #435465;
}

.c4:not(:disabled):not(:active):hover .sc-dlfnbm * {
  fill: #435465;
}

.c4 .sc-dlfnbm + .c5 {
  margin-left: 0.8rem;
}

.c4 .c5 + .sc-dlfnbm {
  margin-left: 0.8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem;
  position: fixed;
  z-index: 1001;
  top: 0;
  right: 0;
}

.c0 > * + * {
  margin-top: 2rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  gap: 1rem;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  min-width: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  gap: 1rem;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  min-width: 100%;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  background-color: #eff2f4;
  overflow: auto;
  padding-top: 32px;
}

@media (max-width:1279px) {
  .c1 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1280px) {
  .c1 {
    padding-left: calc(24px + (((100vw - 1280px) / 160) * 16));
    padding-right: calc(24px + (((100vw - 1280px) / 160) * 16));
  }
}

@media (min-width:1440px) {
  .c1 {
    padding-left: 40px;
    padding-right: 40px;
  }
}

<div>
  <div
    class="c0"
  />
  <main
    class="c1"
  >
    <div
      class="c2"
    />
    <div
      class="c3"
    >
      <button
        class="c4"
        type="button"
      >
        <span
          class="c5 c6"
        >
          Login with Github
        </span>
      </button>
      <button
        class="c4"
        type="button"
      >
        <span
          class="c5 c6"
        >
          Fetch repo
        </span>
      </button>
    </div>
  </main>
</div>
`;

exports[`Dashboard should show loading if isLoading Prop is true 1`] = `
.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c6 {
  border: none;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  padding: 0.8rem 2.4rem 1rem;
  border-radius: 0.8rem;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6,
.c6 .c7,
.c6 .sc-dlfnbm,
.c6 .sc-dlfnbm * {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6:hover {
  cursor: pointer;
}

.c6:focus {
  outline: none;
}

.c6:disabled {
  cursor: not-allowed;
}

.c6::after {
  content: '';
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: transparent;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.8rem;
  border-style: solid;
}

.c6:disabled {
  color: #dfe4e9;
  background-color: transparent;
}

.c6:disabled::after {
  border-width: 0.1rem;
  border-color: #dfe4e9;
}

.c6:disabled .sc-dlfnbm * {
  fill: #dfe4e9;
}

.c6:not(:disabled):not(:hover) {
  color: #303C48;
  background-color: transparent;
}

.c6:not(:disabled):not(:hover)::after {
  border-width: 0.1rem;
  border-color: #303C48;
}

.c6:not(:disabled):not(:hover) .sc-dlfnbm * {
  fill: #303C48;
}

.c6:not(:disabled):active {
  color: #546A7F;
  background-color: transparent;
}

.c6:not(:disabled):active::after {
  border-width: 0.2rem;
  border-color: #546A7F;
}

.c6:not(:disabled):active .sc-dlfnbm * {
  fill: #546A7F;
}

.c6:not(:disabled):not(:active):hover {
  color: #435465;
  background-color: transparent;
  box-shadow: 0 0.2rem 0.8rem rgba(67,84,101,0.35);
}

.c6:not(:disabled):not(:active):hover::after {
  border-width: 0.2rem;
  border-color: #435465;
}

.c6:not(:disabled):not(:active):hover .sc-dlfnbm * {
  fill: #435465;
}

.c6 .sc-dlfnbm + .c7 {
  margin-left: 0.8rem;
}

.c6 .c7 + .sc-dlfnbm {
  margin-left: 0.8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem;
  position: fixed;
  z-index: 1001;
  top: 0;
  right: 0;
}

.c0 > * + * {
  margin-top: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  gap: 1rem;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  min-width: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  gap: 1rem;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  min-width: 100%;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  font-size: 56px;
}

.c3 * {
  fill: #126AFA;
}

.c1 {
  background-color: #eff2f4;
  overflow: auto;
  padding-top: 32px;
}

.c2 {
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  background-color: rgba(255,255,255,0.7);
  z-index: 1;
}

@media (max-width:1279px) {
  .c1 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1280px) {
  .c1 {
    padding-left: calc(24px + (((100vw - 1280px) / 160) * 16));
    padding-right: calc(24px + (((100vw - 1280px) / 160) * 16));
  }
}

@media (min-width:1440px) {
  .c1 {
    padding-left: 40px;
    padding-right: 40px;
  }
}

<div>
  <div
    class="c0"
  />
  <main
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        class="c3"
        height="1em"
        viewBox="0 0 100 100"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M82 35.7C74.1 18 53.4 10.1 35.7 18S10.1 46.6 18 64.3l7.6-3.4c-6-13.5 0-29.3 13.5-35.3s29.3 0 35.3 13.5l7.6-3.4z"
        >
          <animatetransform
            attributeName="transform"
            attributeType="XML"
            dur="2s"
            from="0 50 50"
            repeatCount="indefinite"
            to="360 50 50"
            type="rotate"
          />
        </path>
      </svg>
    </div>
    <div
      class="c4"
    />
    <div
      class="c5"
    >
      <button
        class="c6"
        type="button"
      >
        <span
          class="c7 c8"
        >
          Login with Github
        </span>
      </button>
      <button
        class="c6"
        type="button"
      >
        <span
          class="c7 c8"
        >
          Fetch repo
        </span>
      </button>
    </div>
  </main>
</div>
`;
